

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>MAKER Genome Annotation using cc-tools and Jetstream(WQ-MAKER) &mdash; YOUR QUICKSTART NAME HERE as topic_quick_start 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
        <script src="_static/cyverse.js"></script>
        <script src="_static/intercom-script-for-learning.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/cyverse.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> YOUR QUICKSTART NAME HERE as topic_quick_start
          

          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#"><strong>MAKER Genome Annotation using cc-tools and Jetstream(WQ-MAKER)</strong></a></li>
<li><a class="reference internal" href="#part-3-set-up-a-maker-run-using-the-terminal-window-instead-if-you-want-to-run-the-wq-maker-using-jupyter-notebook-please-see-the-section-below">Part 3: Set up a MAKER run using the Terminal window. Instead IF you want to run the WQ-MAKER using Jupyter notebook please see the section below</a></li>
<li><a class="reference internal" href="#note-advanced-users"><strong>Note(Advanced Users)</strong></a><ul>
<li><a class="reference internal" href="#note"><strong>Note</strong></a></li>
</ul>
</li>
<li><a class="reference internal" href="#moving-data-from-cyverse-datastore-using-icommands"><strong>Moving data from CyVerse Datastore using iCommands</strong></a></li>
</ul>
</div>
            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">YOUR QUICKSTART NAME HERE as topic_quick_start</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li><strong>MAKER Genome Annotation using cc-tools and Jetstream(WQ-MAKER)</strong></li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/Part3.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <p><a class="reference external" href="http://learning.cyverse.org/"><img alt="CyVerse logo" src="_images/cyverse_rgb.png" style="width: 500px; height: 100px;" /></a></p>
<p><a class="reference external" href="http://learning.cyverse.org/"><img alt="Home_Icon" src="_images/homeicon.png" style="width: 25px; height: 25px;" /></a>
<a class="reference external" href="http://learning.cyverse.org/">Learning Center Home</a></p>
<div class="section" id="maker-genome-annotation-using-cc-tools-and-jetstream-wq-maker">
<h1><strong>MAKER Genome Annotation using cc-tools and Jetstream(WQ-MAKER)</strong><a class="headerlink" href="#maker-genome-annotation-using-cc-tools-and-jetstream-wq-maker" title="Permalink to this headline">¶</a></h1>
</div>
<div class="section" id="part-3-set-up-a-maker-run-using-the-terminal-window-instead-if-you-want-to-run-the-wq-maker-using-jupyter-notebook-please-see-the-section-below">
<h1>Part 3: Set up a MAKER run using the Terminal window. Instead IF you want to run the WQ-MAKER using Jupyter notebook please see the section below<a class="headerlink" href="#part-3-set-up-a-maker-run-using-the-terminal-window-instead-if-you-want-to-run-the-wq-maker-using-jupyter-notebook-please-see-the-section-below" title="Permalink to this headline">¶</a></h1>
<p>Step 1:  Navigate to the mounted volume.</p>
<p>Once you have logged in to your instance using webshell or ssh of your MASTER instance, you must change the directory permissions as below</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#Change the ownership and group permission on the mount location</span>
$ sudo chown -hR <span class="nv">$USER</span> /vol_b
$ sudo chgrp -hR <span class="nv">$USER</span> /vol_b

<span class="c1"># cd into the /vol_b and then run WQ-MAKER in there</span>
$ <span class="nb">cd</span> /vol_b
</pre></div>
</div>
<p>Step 2: Get oriented. You will find staged example data in “/opt/WQ-MAKER_example_data/” within the MASTER instance.  List its contents with the ls command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ls /opt/WQ-MAKER_example_data/
maker_bopts.ctl  maker_exe.ctl  maker-hosts  maker_opts.ctl  test_data  worker-launch.yml

$ ls /opt/WQ-MAKER_example_data/test_data
mRNA.fasta  msu-irgsp-proteins.fasta  Os-rRNA.fa  plant_repeats.fasta  test_genome_chr1.fasta  test_genome.fasta
</pre></div>
</div>
<p>2.1 maker_*.ctl file are a set of configuration files that can be used for this exercise or generated as described below.</p>
<p>2.2 worker-launch.yml and maker-hosts are ansible-playbook and host file for luanching jobs on WORKERS (optional for WQ-MAKER)</p>
<p>2.3 fasta files include a scaled-down genome (test_genome.fasta) which is comprised of the first 300kb of 12 chromosomes of rice and scaled-down genome (test_genome_chr1.fasta) which is comprised of the first 300kb of first chromosome of rice</p>
<p>2.4 mRNA sequences from NCBI (mRNA.fasta)</p>
<p>2.5 publicly available annotated protein sequences of rice (MSU7.0 and IRGSP1.0) - msu-irgsp-proteins.fasta</p>
<p>2.6 collection of plant repeats (plant_repeats.fasta)</p>
<p>2.7ribosomal RNAsequence of rice (Os-rRNA.fa)</p>
<p>Executables for running MAKER are located in /opt/maker/bin and /opt/maker/exe:</p>
<p>As the names suggest the “/opt/maker/bin” directory includes many useful auxiliary scripts.  For example cufflinks2gff3 will convert output from an RNA-seq analysis into a GFF3 file that can be used for input as evidence for WQ-MAKER. RepeatMasker, augustus, blast, exonerate, and snap are programs that MAKER uses in its pipeline.  We recommend reading MAKER Tutorial at GMOD for more information about these.</p>
<p>Step 3: Set up a WQ-MAKER run.  Create a working directory called “maker_run” on your home directory using the mkdir command and use cd to move into that directory:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Navigate to the mounted volume for creating test directory</span>
$ <span class="nb">cd</span> /vol_b
$ mkdir wq_maker_run
$ <span class="nb">cd</span> wq_maker_run
</pre></div>
</div>
<p>Step 4: Copy the contents of  “WQ-MAKER_example_data” into the current directory using cp -r command.  Verify using the ls command.  Change the permissions on that directory</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sudo cp -r /opt/WQ-MAKER_example_data/test_data .
</pre></div>
</div>
<p>Step 5: Run the maker command with the –help flag to get a usage statement and list of options:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ maker -h
Argument <span class="s2">&quot;2.53_01&quot;</span> isn<span class="s1">&#39;t numeric in numeric ge (&gt;=) at /usr/local/lib/x86_64-linux-gnu/perl/5.22.1/forks.pm line 1570.</span>
<span class="s1">MAKER version 2.31.9</span>
<span class="s1">Usage:</span>
<span class="s1"> maker [options] &lt;maker_opts&gt; &lt;maker_bopts&gt; &lt;maker_exe&gt;</span>
<span class="s1">Description:</span>
<span class="s1"> MAKER is a program that produces gene annotations in GFF3 format using</span>
<span class="s1"> evidence such as EST alignments and protein homology. MAKER can be used to</span>
<span class="s1"> produce gene annotations for new genomes as well as update annotations</span>
<span class="s1"> from existing genome databases.</span>
<span class="s1"> The three input arguments are control files that specify how MAKER should</span>
<span class="s1"> behave. All options for MAKER should be set in the control files, but a</span>
<span class="s1"> few can also be set on the command line. Command line options provide a</span>
<span class="s1"> convenient machanism to override commonly altered control file values.</span>
<span class="s1"> MAKER will automatically search for the control files in the current</span>
<span class="s1"> working directory if they are not specified on the command line.</span>
<span class="s1"> Input files listed in the control options files must be in fasta format</span>
<span class="s1"> unless otherwise specified. Please see MAKER documentation to learn more</span>
<span class="s1"> about control file  configuration.  MAKER will automatically try and</span>
<span class="s1"> locate the user control files in the current working directory if these</span>
<span class="s1"> arguments are not supplied when initializing MAKER.</span>
<span class="s1"> It is important to note that MAKER does not try and recalculated data that</span>
<span class="s1"> it has already calculated.  For example, if you run an analysis twice on</span>
<span class="s1"> the same dataset you will notice that MAKER does not rerun any of the</span>
<span class="s1"> BLAST analyses, but instead uses the blast analyses stored from the</span>
<span class="s1"> previous run. To force MAKER to rerun all analyses, use the -f flag.</span>
<span class="s1"> MAKER also supports parallelization via MPI on computer clusters. Just</span>
<span class="s1"> launch MAKER via mpiexec (i.e. mpiexec -n 40 maker). MPI support must be</span>
<span class="s1"> configured during the MAKER installation process for this to work though</span>
<span class="s1">Options:</span>
<span class="s1"> -genome|g &lt;file&gt;    Overrides the genome file path in the control files</span>
<span class="s1"> -RM_off|R           Turns all repeat masking options off.</span>
<span class="s1"> -datastore/         Forcably turn on/off MAKER&#39;</span>s two deep directory
  nodatastore        structure <span class="k">for</span> output.  Always on by default.
 -old_struct         Use the old directory styles <span class="o">(</span>MAKER <span class="m">2</span>.26 and lower<span class="o">)</span>
 -base    &lt;string&gt;   Set the base name MAKER uses to save output files.
                     MAKER uses the input genome file name by default.
 -tries<span class="p">|</span>t &lt;integer&gt;  Run contigs up to the specified number of tries.
 -cpus<span class="p">|</span>c  &lt;integer&gt;  Tells how many cpus to use <span class="k">for</span> BLAST analysis.
                     Note: this is <span class="k">for</span> BLAST and not <span class="k">for</span> MPI!
 -force<span class="p">|</span>f            Forces MAKER to delete old files before running again.
                     This will require all blast analyses to be rerun.
 -again<span class="p">|</span>a            recaculate all annotations and output files even <span class="k">if</span> no
                     settings have changed. Does not delete old analyses.
 -quiet<span class="p">|</span>q            Regular quiet. Only a handlful of status messages.
 -qq                 Even more quiet. There are no status messages.
 -dsindex            Quickly generate datastore index file. Note that this
                     will not check <span class="k">if</span> run settings have changed on contigs
 -nolock             Turn off file locks. May be usful on some file systems,
                     but can cause race conditions <span class="k">if</span> running in parallel.
 -TMP                Specify temporary directory to use.
 -CTL                Generate empty control files in the current directory.
 -OPTS               Generates just the maker_opts.ctl file.
 -BOPTS              Generates just the maker_bopts.ctl file.
 -EXE                Generates just the maker_exe.ctl file.
 -MWAS    &lt;option&gt;   Easy way to control mwas_server <span class="k">for</span> web-based GUI
                          options:  STOP
                                    START
                                    RESTART
 -version            Prints the MAKER version.
 -help<span class="p">|</span>?             Prints this usage statement.
</pre></div>
</div>
<p>Step 6: Create control files that tell MAKER what to do. Three files are required:</p>
<blockquote>
<div><p>6.1 maker_opts.ctl - gives location of input files (genome and evidence) and sets options that affect MAKER behavior</p>
<p>6.2 maker_exe.ctl - gives path information for the underlying executables.</p>
<p>6.3 maker_bopt.ctl - sets parameters for filtering BLAST and Exonerate alignment results</p>
</div></blockquote>
<p>To create these files run the maker command with the -CTL flag. Verify with ls:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>   $ maker -CTL
   $ ls
   maker_bopts.ctl  maker_exe.ctl  maker_opts.ctl  test_data

<span class="m">6</span>.4 The <span class="s2">&quot;maker_exe.ctl&quot;</span> is automatically generated with the correct paths to executables and does not need to be modified.

<span class="m">6</span>.5 The <span class="s2">&quot;maker_bopt.ctl&quot;</span> is automatically generated with reasonable default parameters and also does not need to be modified unless you want to experiment with optimization of these parameters.

<span class="m">6</span>.6 The automatically generated <span class="s2">&quot;maker_opts.ctl&quot;</span> file needs to be modified in order to specify the genome file and evidence files to be used as input.  You can use the text editor <span class="s2">&quot;vi&quot;</span> or <span class="s2">&quot;nano&quot;</span> that is already installed in the MASTER instance
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ rm maker_opts.ctl
$ cp /opt/WQ-MAKER_example_data/maker_opts.ctl .
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#-----Gene Prediction</span>
<span class="nv">snaphmm</span><span class="o">=</span> <span class="c1">#SNAP HMM file</span>
<span class="nv">gmhmm</span><span class="o">=</span> <span class="c1">#GeneMark HMM file</span>
<span class="nv">augustus_species</span><span class="o">=</span> <span class="c1">#Augustus gene prediction species model</span>
<span class="nv">fgenesh_par_file</span><span class="o">=</span> <span class="c1">#FGENESH parameter file</span>
<span class="nv">pred_gff</span><span class="o">=</span> <span class="c1">#ab-initio predictions from an external GFF3 file</span>
<span class="nv">model_gff</span><span class="o">=</span> <span class="c1">#annotated gene models from an external GFF3 file (annotation pass-through)</span>
<span class="nv">est2genome</span><span class="o">=</span><span class="m">1</span> <span class="c1">#infer gene predictions directly from ESTs, 1 = yes, 0 = no # Change 1 to 0</span>
<span class="nv">protein2genome</span><span class="o">=</span><span class="m">1</span> <span class="c1">#infer predictions from protein homology, 1 = yes, 0 = no # Change 1 to 0</span>
<span class="nv">unmask</span><span class="o">=</span><span class="m">0</span> <span class="c1">#also run ab-initio prediction programs on unmasked sequence, 1 = yes, 0 = no</span>
</pre></div>
</div>
<p>Otherwise open the maker_opts.ctl in a text editor of choice )</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ nano maker_opts.ctl
</pre></div>
</div>
<p>Here are the sections of the “maker_opts.ctl” file you need to edit.  For more information about the this please check this The_MAKER_control_files_explained - Add path information to files as shown.</p>
<p>This section pertains to specifying the genome assembly to be annotated and setting organism type:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#-----Genome (these are always required)</span>
<span class="nv">genome</span><span class="o">=</span>./test_data/test_genome.fasta <span class="c1">#genome sequence (fasta file or fasta embeded in GFF3 file)</span>
<span class="nv">organism_type</span><span class="o">=</span>eukaryotic <span class="c1">#eukaryotic or prokaryotic. Default is eukaryotic</span>
</pre></div>
</div>
<p>The following section pertains to EST and other mRNA expression evidence.  Here we are only using same species data, but one could specify data from a related species using the “altest” parameter. With RNA-seq data aligned to your genome by Cufflinks or Tophat one could use maker auxiliary scripts (cufflinks2gff3 and tophat2gff3) to generate GFF3 files and specify these using the est_gff parameter:</p>
<p>..code-block:: bash</p>
<blockquote>
<div><p>#—–EST Evidence (for best results provide a file for at least one)
est=./test_data/mRNA.fasta #set of ESTs or assembled mRNA-seq in fasta format
altest= #EST/cDNA sequence file in fasta format from an alternate organism
est_gff= #aligned ESTs or mRNA-seq from an external GFF3 file
altest_gff= #aligned ESTs from a closely relate species in GFF3 format</p>
</div></blockquote>
<p>The following section pertains to protein sequence evidence.  Here we are using previously annotated protein sequences.  Another option would be to use SwissProt or other database:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#-----Protein Homology Evidence (for best results provide a file for at least one)</span>
<span class="nv">protein</span><span class="o">=</span>./test_data/msu-irgsp-proteins.fasta  <span class="c1">#protein sequence file in fasta format (i.e. from mutiple oransisms)</span>
<span class="nv">protein_gff</span><span class="o">=</span>  <span class="c1">#aligned protein homology evidence from an external GFF3 file</span>
</pre></div>
</div>
<p>This next section pertains to repeat identification:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#-----Repeat Masking (leave values blank to skip repeat masking)</span>
<span class="nv">model_org</span><span class="o">=</span> <span class="c1">#select a model organism for RepBase masking in RepeatMasker</span>
<span class="nv">rmlib</span><span class="o">=</span>./test_data/plant_repeats.fasta <span class="c1">#provide an organism specific repeat library in fasta format for RepeatMasker</span>
<span class="nv">repeat_protein</span><span class="o">=</span> <span class="c1">#provide a fasta file of transposable element proteins for RepeatRunner</span>
<span class="nv">rm_gff</span><span class="o">=</span> <span class="c1">#pre-identified repeat elements from an external GFF3 file</span>
<span class="nv">prok_rm</span><span class="o">=</span><span class="m">0</span> <span class="c1">#forces MAKER to repeatmask prokaryotes (no reason to change this), 1 = yes, 0 = no</span>
<span class="nv">softmask</span><span class="o">=</span><span class="m">1</span> <span class="c1">#use soft-masking rather than hard-masking in BLAST (i.e. seg and dust filtering)</span>
</pre></div>
</div>
<p>Step 7: Run WQ-MAKER</p>
<p>Before running MAKER, check to make sure all worker instances have become active.</p>
<p>On the MASTER instance, make sure you are in the “maker_run” directory and all of your files are in place and then run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ nohup wq_maker -contigs-per-split <span class="m">1</span> -cores <span class="m">1</span> -memory <span class="m">2048</span> -disk <span class="m">4096</span> -N wq_test_<span class="si">${</span><span class="nv">USER</span><span class="si">}</span> -d all -o master.dbg -debug_size_limit<span class="o">=</span><span class="m">0</span> -stats test_out_stats.txt &gt; log_file.txt <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
</pre></div>
</div>
<p>7.1 -contigs-per-split 1: splits the genome file into 1 contig/scaffold/sequence per file. By specifiying this option, we are telling wq_maker to split the genome file into 1 sequence per file. By default, the wq_maker splits the fasta file into 10 sequences per file and this case, it is not ideal because, there will be 2 files (1 containing chromosomes from 1-10 and the other containing 11-12). This will decrease the speed at the wq_maker annotates the genome.</p>
<p><strong>Warning</strong>: Unless otherwise you have a complete genome containing chromosomes or very few scaffolds, it is not recommended to use this option. For example if you have a genome that contains 10,000 sequences, then this option will create 10,000 files on your working directory which is not ideal of navigation purposes. Check to see how many contigs/scaffolds/chromosomes you have in your genome using <cite>grep “&gt;” -c &lt;genome fasta file&gt;</cite> and if the number is too high, then avoid this option</p>
<p>7.2 N maker_run_ud sets the project name to wq_test_{USER}. This is mandatory if we need to run WQ-MAKER.</p>
<p>7.3 -d all Sets the debug flag for Work Queue. For all debugging output, try ‘all’
7.4 -o master.dbg Sets the debug file for Work Queue
7.5 -debug_size_limit=0 Sets the byte wrap around on the debug file. 0 signifies it is never wrapped (Default it 1M)</p>
<p>Wait for the MASTER to advertise master status to the catalog server before your run WQ-MAKER on the WORKERS (see below).</p>
<p><strong>INFO</strong>
$ tail log_file.txt</p>
<p>Mon Sep 11 15:08:22 2017 :: Submitting file ./test_data/test_genome.fasta_000008 for processing.
Mon Sep 11 15:08:22 2017 :: Submitted task 11 for annotating ./test_data/test_genome.fasta_000008 with command: mpiexec -n 1 maker -g ./test_data/test_genome.fasta_000008 -base test_genome -debug_size_limit=0
Mon Sep 11 15:08:22 2017 :: Submitting file ./test_data/test_genome.fasta_000006 for processing.
Mon Sep 11 15:08:22 2017 :: Submitted task 12 for annotating ./test_data/test_genome.fasta_000006 with command: mpiexec -n 1 maker -g ./test_data/test_genome.fasta_000006 -base test_genome -debug_size_limit=0
warning: this work queue master is visible to the public.
warning: you should set a password with the –password option.</p>
<p>Once the log_file show the above output and once your WORKERS are in active state, then either ssh or use webshell into each of the WORKERS and then run</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ nohup work_queue_worker -N wq_test_<span class="si">${</span><span class="nv">USER</span><span class="si">}</span> --cores all --debug-rotate-max<span class="o">=</span><span class="m">0</span> -d all -o worker.dbg &gt; log_file_2.txt <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
</pre></div>
</div>
<p>7.6 -N wq_test_${USER} sets the project name to maker_run_test. This is mandatory if we need to run WQ-MAKER. This is the same id that we have specified with MASTER</p>
<p>7.7 –debug-rotate-max=0 Set the maximum size of the debug log (default 10M, 0 disables)</p>
<p>7.8 -d all Sets the debug flag for Work Queue. For all debugging output, try ‘all’
7.9 -o worker.dbg Sets the debug file for Work Queue</p>
<p>7.10 –cores all Uses all the cores on the machine</p>
</div>
<div class="section" id="note-advanced-users">
<h1><strong>Note(Advanced Users)</strong><a class="headerlink" href="#note-advanced-users" title="Permalink to this headline">¶</a></h1>
<p>You can use Anisble method to launch jobs without ssh’ing into WORKERS from the MASTER itself. Once the maker run is started on the master, and once your WORKERS are in active state</p>
<p>Step 1: Copy ansible.cfg file into your home directory which will help you to avoid host verification</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ cp /opt/WQ-MAKER_example_data/.ansible.cfg ~
</pre></div>
</div>
<p>Step 2: Add ssh keys of MASTER to the JetStream atmosphere. This will allow Ansible to launch the jobs without ssh into the WORKERS.
Step 3: Copy maker-hosts file into your working directory and populate it with ip addresses of the WORKERS</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ cp /opt/WQ-MAKER_example_data/maker-hosts .
$ <span class="nb">echo</span> <span class="s2">&quot;149.165.169.21&quot;</span> &gt;&gt; maker-hosts <span class="c1"># This ip address of the WORKER is specific to my account. This will not work for you</span>
$ <span class="nb">echo</span> <span class="s2">&quot;149.165.169.78&quot;</span> &gt;&gt; maker-hosts <span class="c1"># This ip address of the WORKER is specific to my account. This will not work for you</span>
</pre></div>
</div>
<p>Step 4: Copy the Ansible playbook to your working directory.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>   $ cp /opt/WQ-MAKER_example_data/worker-launch.yml .
   $ cat worker-launch.yml

   ---
   - hosts : workers
     environment:
       PATH: <span class="s2">&quot;{{ ansible_env.PATH }}:/home/</span><span class="si">${</span><span class="nv">USER</span><span class="si">}</span><span class="s2">/bin:/home/</span><span class="si">${</span><span class="nv">USER</span><span class="si">}</span><span class="s2">/.local/bin:/opt/icommands:/opt/icommands:/opt/exonerate-2.2.0-x86_64/bin/:/opt/cctools/bin:/opt/ncbi-blast-2.6.0+/bin/:/opt/snoscan-0.9.1/:/opt/tRNAscan-SE-1.3.1/:/opt/snap/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/augustus-3.2.2/bin:/opt/maker/bin:/opt/RepeatMasker:/opt/snap&quot;</span>
       PERL5LIB: <span class="s2">&quot;/opt/tRNAscan-SE-1.3.1::/opt/cctools/lib/perl5/site_perl&quot;</span>
     tasks :
     - name : Execute the script
       shell : /opt/cctools/bin/work_queue_worker -N wq_test_<span class="si">${</span><span class="nv">USER</span><span class="si">}</span> -s /home/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span> --cores all --debug-rotate-max<span class="o">=</span><span class="m">0</span> -d all -o /home/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span>/worker.dbg

<span class="m">4</span>.1 - hosts is the name of the hosts <span class="o">(</span>workers in this <span class="k">case</span>. It can be anything<span class="o">)</span>
<span class="m">4</span>.2 tasks is the task that need to be performed by the Ansible <span class="o">(</span>In this <span class="k">case</span> run work_queue_worker<span class="o">)</span>
<span class="m">4</span>.3 name is just name of the task <span class="o">(</span>It can be anything<span class="o">)</span>
<span class="m">4</span>.4 -N maker_run_test sets the project name to maker_run_test. This is mandatory <span class="k">if</span> we need to run WQ-MAKER
<span class="m">4</span>.5 -s /home/upendra/ Set the location <span class="k">for</span> creating the working directory of the worker
<span class="m">4</span>.6 --debug-rotate-max<span class="o">=</span><span class="m">0</span> Set the maximum size of the debug log <span class="o">(</span>default 10M, <span class="m">0</span> disables<span class="o">)</span>
<span class="m">4</span>.7 -d all Sets the debug flag <span class="k">for</span> Work Queue. For all debugging output, try <span class="s1">&#39;all&#39;</span>
<span class="m">4</span>.8 -o worker.dbg Sets the debug file <span class="k">for</span> Work Queue
</pre></div>
</div>
<p>Step 5: Run WQ-MAKER on the WORKERS</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ nohup ansible-playbook -u <span class="si">${</span><span class="nv">USER</span><span class="si">}</span> -i maker-hosts worker-launch.yml &gt; log_file_2.txt <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
</pre></div>
</div>
<p>To check the status of the WQ-MAKER job, run the following.
.. code-block:: bash</p>
<blockquote>
<div><p>$ work_queue_status -M wq_test_${USER}
PROJECT            HOST                   PORT WAITING RUNNING COMPLETE WORKERS
maker_run_test     js-157-131.jetstream-  9155       8       4        0       4</p>
</div></blockquote>
<p>Step 8. Stats output from MASTER instance
The log_file.txt will tell you if the job has been finished or not.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ tail log_file.txt

WQ-MAKER Start_time:    <span class="m">1505157588000000</span>
WQ-MAKER End_time:      <span class="m">1505157849000000</span>
WQ-MAKER Elapsed:       0d <span class="m">0</span>:04:21.000000
Work Queue Wall Time:   0d <span class="m">0</span>:04:00.427755
Cumulative Task Wall Time:      0d <span class="m">0</span>:36:25.377304
Cumulative Task Good Execute Time:      0d <span class="m">0</span>:36:25.377304
Work Queue Send Time:   0d <span class="m">0</span>:00:01.437632
Work Queue Receive Time:        0d <span class="m">0</span>:00:03.863163
-----------------------------------------------------------------
Mon Sep <span class="m">11</span> <span class="m">15</span>:24:09 <span class="m">2017</span> :: MPI used :: Cores <span class="m">1</span> :: Memory <span class="m">1024</span> :: Disk <span class="m">2048</span>
-----------------------------------------------------------------
</pre></div>
</div>
<p>The following are the output files from WQ-MAKER</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ls test_genome.maker.output/test_genome.maker.output
maker_bopts.log  maker_exe.log  maker_opts.log  mpi_blastdb  test_genome_datastore  test_genome_master_datastore_index.log
</pre></div>
</div>
<p>8.1 The maker_opts.log, maker_exe.log, and maker_bopts.log files are logs of the control files used for this run of MAKER.
8.2 The mpi_blastdb directory contains FASTA indexes and BLAST database files created from the input EST, protein, and repeat databases.
8.3 test_genome_master_datastore_index.log contains information on both the run status of individual contigs and information on where individual contig data is stored.
8.4 The test_genome_datastore directory contains a set of subfolders, each containing the final MAKER output for individual contigs from the genomic fasta file.</p>
<p>Check the test_genome_master_datastore_index.log and task_outputs.txt to see if there were any failures:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ cat test_genome.maker.output/test_genome_master_datastore_index.log
Chr1    test_genome_datastore/41/30/Chr1/       STARTED
Chr10   test_genome_datastore/7C/72/Chr10/      STARTED
Chr11   test_genome_datastore/1E/AA/Chr11/      STARTED
Chr12   test_genome_datastore/1B/FA/Chr12/      STARTED
Chr2    test_genome_datastore/E9/36/Chr2/       STARTED
Chr3    test_genome_datastore/CC/EF/Chr3/       STARTED
Chr4    test_genome_datastore/A3/11/Chr4/       STARTED
Chr5    test_genome_datastore/8A/9B/Chr5/       STARTED
Chr6    test_genome_datastore/13/44/Chr6/       STARTED
Chr7    test_genome_datastore/91/B7/Chr7/       STARTED
Chr8    test_genome_datastore/9A/9E/Chr8/       STARTED
Chr9    test_genome_datastore/87/90/Chr9/       STARTED
Chr1    test_genome_datastore/41/30/Chr1/       FINISHED
Chr10   test_genome_datastore/7C/72/Chr10/      FINISHED
Chr11   test_genome_datastore/1E/AA/Chr11/      FINISHED
Chr12   test_genome_datastore/1B/FA/Chr12/      FINISHED
Chr2    test_genome_datastore/E9/36/Chr2/       FINISHED
Chr3    test_genome_datastore/CC/EF/Chr3/       FINISHED
Chr4    test_genome_datastore/A3/11/Chr4/       FINISHED
Chr5    test_genome_datastore/8A/9B/Chr5/       FINISHED
Chr6    test_genome_datastore/13/44/Chr6/       FINISHED
Chr7    test_genome_datastore/91/B7/Chr7/       FINISHED
Chr8    test_genome_datastore/9A/9E/Chr8/       FINISHED
Chr9    test_genome_datastore/87/90/Chr9/       FINISHED
</pre></div>
</div>
<p>All completed.  Other possible status entries include:</p>
<ul class="simple">
<li><p>FAILED - indicates a failed run on this contig, MAKER will retry these</p></li>
<li><p>RETRY - indicates that MAKER is retrying a contig that failed</p></li>
<li><p>SKIPPED_SMALL - indicates the contig was too short to annotate (minimum contig length is specified in maker_opt.ctl)</p></li>
<li><p>DIED_SKIPPED_PERMANENT - indicates a failed contig that MAKER will not attempt to retry (number of times to retry a contig is specified in maker_opt.ctl)</p></li>
</ul>
<p>The actual output data is stored in in nested set of directories under* test_genome_datastore* in a nested directory structure.</p>
<p>A typical set of outputs for a contig looks like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ls test_genome.maker.output/test_genome_datastore/*/*/*
test_genome_datastore/13/44/Chr6:
Chr6.gff                                                Chr6.maker.proteins.fasta                 Chr6.maker.transcripts.fasta
Chr6.maker.non_overlapping_ab_initio.proteins.fasta     Chr6.maker.snap_masked.proteins.fasta     run.log
Chr6.maker.non_overlapping_ab_initio.transcripts.fasta  Chr6.maker.snap_masked.transcripts.fasta  theVoid.Chr6
</pre></div>
</div>
<ul class="simple">
<li><p>The Chr6.gff file is in GFF3 format and contains the maker gene models and underlying evidence such as repeat regions, alignment data, and ab initio gene predictions, as well as fasta sequence.  Having all of these data in one file is important to enable visualization of the called gene models and underlying evidence, especially using tools like Apollo which enable manual editing and curation of gene models.</p></li>
<li><p>The fasta files Chr6.maker.proteins.fasta and Chr6.maker.transcripts.fasta contain the protein and transcript sequences for the final MAKER gene calls.</p></li>
<li><p>The Chr6.maker.non_overlapping_ab_initio.proteins.fasta and Chr6.maker.non_overlapping_ab_initio.transcripts.fasta files are models that don’t overlap MAKER genes that were rejected for lack of support.</p></li>
<li><p>The Chr6.maker.snap_masked.proteins.fasta and Chr6.maker.snap_masked.transcript.fasta are the initial SNAP predicted models not further processed by MAKER</p></li>
</ul>
<p>The output directory theVoid.Chr1 contains raw output data from all of the pipeline steps.  One useful file found here is the repeat-masked version of the contig, query.masked.fasta.</p>
<p>Step 9: Merge the gff files</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ gff3_merge -n -d test_genome.maker.output/test_genome_master_datastore_index.log
</pre></div>
</div>
<p>9.1 -d The location of the MAKER datastore index log file.
9.2 -n Do not print fasta sequence in footer</p>
<div class="section" id="note">
<h2><strong>Note</strong><a class="headerlink" href="#note" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p>By default, the output of the gff3_merge is test_genome.all.gff, but you can have an alternate base name for the output files using “-o” option</p>
<p>If you want to perform abinition gene predictions then you should skip <cite>-n</cite> option. Run</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ gff3_merge -d test_genome_master_datastore_index.log
</pre></div>
</div>
<p>And follow the rest of the steps of abinitio gene predictions in here</p>
<p>The final output from gff3_merge is “test_genome.all.gff”</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">##gff-version 3</span>
Chr6    maker   gene    <span class="m">43764</span>   <span class="m">46139</span>   .   -   .   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3<span class="p">;</span><span class="nv">Name</span><span class="o">=</span>maker-Chr6-snap-gene-0.3
Chr6    maker   mRNA    <span class="m">43764</span>   <span class="m">46139</span>   .   -   .   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3<span class="p">;</span><span class="nv">Name</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1<span class="p">;</span><span class="nv">_AED</span><span class="o">=</span><span class="m">0</span>.12<span class="p">;</span><span class="nv">_eAED</span><span class="o">=</span><span class="m">0</span>.50<span class="p">;</span><span class="nv">_QI</span><span class="o">=</span><span class="m">64</span><span class="p">|</span><span class="m">0</span><span class="p">|</span><span class="m">0</span><span class="p">|</span><span class="m">1</span><span class="p">|</span><span class="m">0</span><span class="p">|</span><span class="m">0</span>.33<span class="p">|</span><span class="m">3</span><span class="p">|</span><span class="m">0</span><span class="p">|</span><span class="m">76</span>
Chr6    maker   exon    <span class="m">43764</span>   <span class="m">43846</span>   .   -   .   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1:exon:2<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1
Chr6    maker   exon    <span class="m">44833</span>   <span class="m">44896</span>   .   -   .   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1:exon:1<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1
Chr6    maker   exon    <span class="m">45992</span>   <span class="m">46139</span>   .   -   .   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1:exon:0<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1
Chr6    maker   five_prime_UTR  <span class="m">46076</span>   <span class="m">46139</span>   .   -   .   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1:five_prime_utr<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1
Chr6    maker   CDS <span class="m">45992</span>   <span class="m">46075</span>   .   -   <span class="m">0</span>   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1:cds<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1
Chr6    maker   CDS <span class="m">44833</span>   <span class="m">44896</span>   .   -   <span class="m">0</span>   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1:cds<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1
Chr6    maker   CDS <span class="m">43764</span>   <span class="m">43846</span>   .   -   <span class="m">2</span>   <span class="nv">ID</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1:cds<span class="p">;</span><span class="nv">Parent</span><span class="o">=</span>maker-Chr6-snap-gene-0.3-mRNA-1
</pre></div>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="moving-data-from-cyverse-datastore-using-icommands">
<h1><strong>Moving data from CyVerse Datastore using iCommands</strong><a class="headerlink" href="#moving-data-from-cyverse-datastore-using-icommands" title="Permalink to this headline">¶</a></h1>
<p>iCommands is a collection of commands for Linux and Mac OS operating systems that are used in the iRODS system to interact with the CyVerse Data Store. Many commands are very similar to Unix utilities. For example, to list files and directories, in Linux you use ls, but in iCommands you use ils.
While iCommands are great for all transfers and for automating tasks via scripts, they are the best choice for large files (2-100 GB each) and for bulk file transfers (many small files). For a comparison of the different methods of uploading and downloading data items, see Downloading and Uploading Data.
iCommands can be used by CyVerse account users to download files that have been shared by other users and to upload files to the Data Store, as well as add metadata, change permissions, and more. Commonly used iCommands are listed below. Follow the instructions on Setting Up iCommands for how to download and configure iCommands for your operating system.
A CyVerse account is not required to download a public data file via iCommands. To see instructions just for public data download with iCommands, see the iCommands section on Downloading Data Files Without a User Account.
For configuring icommands and the different commands that can be used to move the data in and out of datastore, please refer this link. You may want to watch a CyVerse video about iCommands.</p>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, CyVerse

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>